File: Java-Core/src/main/java/prompto/value/NativeResource.java
package prompto.value;

import java.io.IOException;

import prompto.declaration.NativeResourceDeclaration;
import prompto.error.SyntaxError;

public class NativeResource extends NativeInstance implements IResource {

	public NativeResource(NativeResourceDeclaration declaration) throws SyntaxError {
		super(declaration);
	}
	
	@Override
	public boolean isReadable() {
		return ((IResource)instance).isReadable();
	}
	
	@Override
	public boolean isWritable() {
		return ((IResource)instance).isWritable();
	}
	
	@Override
	public String readFully() throws IOException {
		return ((IResource)instance).readFully();
	}
	
	@Override
	public void writeFully(String data) throws IOException {
		((IResource)instance).writeFully(data);
	}
	
	@Override
	public void close() {
		((IResource)instance).close();
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/value/NativeResource.js
var NativeInstance = require("./NativeInstance").NativeInstance;

function NativeResource(declaration) {
	NativeInstance.call(this, declaration);
	return this;
}

NativeResource.prototype = Object.create(NativeInstance.prototype);
NativeResource.prototype.constructor = NativeResource;

NativeResource.prototype.isReadable = function() {
	return this.instance.isReadable();
};

NativeResource.prototype.isWritable = function() {
	return this.instance.isWritable();
};

NativeResource.prototype.readFully = function() {
	return this.instance.readFully();
};

NativeResource.prototype.writeFully = function(data) {
	this.instance.writeFully(data);
};

NativeResource.prototype.close = function() {
	this.instance.close();
};

exports.NativeResource = NativeResource;
